packages/next/src/client/app-index.tsx

```diff
@@ -136,6 +136,10 @@ const DOMContentLoaded = function () {
     initialServerDataWriter.close()
     initialServerDataFlushed = true
     initialServerDataBuffer = undefined
+
+    // Also clear the global data array to release memory. Otherwise it will be held indefinitely.
+    delete self.__next_f
+    nextServerDataLoadingGlobal.length = 0
```

Comment:
I think this is safe b/c we should have already consumed the chunks but seems cleaner to just clear these as soon as we consume them when the module loads rather than waiting for the DOMContentLoaded event

Reply:
Good point!